src/Turtle/Prelude.hs

```diff
@@ -1513,12 +1514,16 @@ appendonly file = using (managed (Filesystem.withTextFile file IO.AppendMode))
 cat :: [Shell a] -> Shell a
 cat = msum
 
+grepWith :: (b -> Text) -> Pattern a -> Shell b -> Shell b
+grepWith f pattern s = mfilter (not . null . match pattern . f) s
```

Comment:
Was there a reason to *not* use `mfilter` before this?

Reply:
No reason.  `mfilter` is just fine :)